sourcegraph/sourcegraph
# [Search UI]: Add new experimental search UI navigation

Closes https://github.com/sourcegraph/sourcegraph/issues/54794

## Problem 
The problem this PR is trying to solve is optimising/freeing vertical space (free it for actual content that matters but still manage our global navigation in a way that this is still visible and accessible). Vertical space becomes a bigger problem in places where we have a lot of sticky content on top (like blob UI page or even search result page), users who don't use big external monitors can't use Sourcegraph efficiently. 

## Solution
The solution is to merge search box UI and global navigation UI, similar how our code search product have it (see Google Code Search and Github navigation). 

<img width="300" alt="Screenshot 2023-11-07 at 13 51 05" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/5c2633ff-c8fe-4160-aecc-40dfec9bbbda"> <img width="300" alt="Screenshot 2023-11-07 at 13 51 07" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/86c926ca-a8e4-48c8-be62-054949df75a1"> 

<img width="300" alt="Screenshot 2023-11-07 at 13 51 18" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/5f1708e1-2343-408a-bb95-1315a47b65e4"> <img width="300" alt="Screenshot 2023-11-07 at 13 51 20" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/bdba124e-0a21-497b-874e-d526d6e2203b">

Note: that we still not sure about some decisions about this navigation, this might affect our feature visibility in unexpected way, so on pages where search isn't related directly (like code insights, batch changes we still show old global navigation UI) 

## Availability 
New navigation UI is under experimental feature flag `newSearchNavigationUI`.

## Other improvements 
- Fixed compact/full navigation transition breakpoint (now navigation switches only and only when there is no enough space for navigation items, previously it was static break point which didn't work properly for generic content)
- Search UI box now has plain query input fallback UI to show when we load search box UI component (it should help with UI thrashing) 

## Further steps 
It's in experimental stage, so implementation may lack some things in layout like Mobile layout might be weird, we will improve it in a follow up PR if this UI will work out.  

## Test plan
- First of all check old navigation, it should be no regression on mobile and desktop versions
- Turn on experimental navigation with experimentalFeatures: `newSearchNavigationUI`
- Check that you can go through search core flow with new navigation.
